File: android/jsbridge/src/main/java/net/kibotu/jsbridge/commands/bottomnavigation/BottomNavigationCommand.kt
package net.kibotu.jsbridge.commands.bottomnavigation

import net.kibotu.jsbridge.SafeAreaService
import net.kibotu.jsbridge.commands.utils.BridgeParsingUtils
import net.kibotu.jsbridge.commands.utils.BridgeResponseUtils
import net.kibotu.jsbridge.commands.BridgeCommand
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.json.JSONObject
import timber.log.Timber

class BottomNavigationCommand(
    private val getBridge: () -> net.kibotu.jsbridge.JavaScriptBridge?
) : BridgeCommand {

    override val action = "bottomNavigation"

    override suspend fun handle(content: Any?): JSONObject = withContext(Dispatchers.Main) {
        try {
            val isVisible = BridgeParsingUtils.parseBoolean(content, "isVisible")

            Timber.i("[handle] isVisible=$isVisible")

            BottomNavigationService.setVisible(isVisible == true)

            SafeAreaService.pushTobridge(getBridge())

            BridgeResponseUtils.createSuccessResponse()
        } catch (e: Exception) {
            Timber.e(e)
            BridgeResponseUtils.createErrorResponse(
                "BOTTOM_NAVIGATION_FAILED",
                e.message ?: "Failed to configure bottom navigation"
            )
        }
    }
}

File: ios/JSBridge/Sources/JSBridge/Commands/bottomnavigation/BottomNavigationCommand.swift
import Foundation

public class BottomNavigationCommand: BridgeCommand {
    public let action = "bottomNavigation"

    weak var bridge: JavaScriptBridge?

    public init(bridge: JavaScriptBridge? = nil) {
        self.bridge = bridge
    }

    @MainActor
    public func handle(content: [String: Any]?) async throws -> [String: Any]? {
        guard let isVisible = content?["isVisible"] as? Bool else {
            throw BridgeError.invalidParameter("isVisible")
        }

        BottomNavigationService.shared.setVisible(isVisible)
        SafeAreaService.shared.pushToBridge(bridge)
        return nil
    }
}

